libgadget/treewalk.c

```diff
@@ -608,7 +610,7 @@ treewalk_run(TreeWalk * tw, int * active_set, int size)
     if(tw->visit) {
         do
         {
-            ev_primary(tw); /* do local particles and prepare export list */
+            ev_primary(tw, NTask); /* do local particles and prepare export list */
```

Comment:
NTask looks a bit out of place here.

What if you add NTask into tw, and then add a local const alias to the variable in the struct?

Reply:
How about const int NTasks = NTask, so we just reuse the current global?

Reply:
and then there is no extra argument